src/dune_rules/odoc.ml

```diff
@@ -443,6 +461,10 @@ let odocs sctx target =
         let odoc = Obj_dir.Module.odoc obj_dir m in
         create_odoc ctx ~target odoc :: acc)
 
+let filter_out_hidden (odoc_files : odoc list) =
+  List.filter odoc_files ~f:(fun odoc_file ->
+      not (is_hidden (Path.Build.to_string odoc_file.html_dir)))
```

Comment:
Rather than determining this from the path, you can just modify the `odocs` function to only use the modules present in `Modules.entry_list`.

Reply:
`Modules.entry_list` seems not to be a good call here, reason being, IIUC, it only contains packages( for instance `foo.opam`) as entry modules, rather than the wrapping module(`foo/Foo`) and the explicit modules (`.ml`) files.

For example, Using only modules present in `Modules.entry_list` results to:
<img width="306" alt="Screenshot 2020-10-31 at 00 25 19" src="https://user-images.githubusercontent.com/34124177/97763866-f7c7cc80-1b1d-11eb-9ef9-d633eaeb5a20.png">

Yet, what we need is sort of:
<img width="387" alt="Screenshot 2020-10-31 at 00 26 19" src="https://user-images.githubusercontent.com/34124177/97763908-17f78b80-1b1e-11eb-9870-efaca15fdfe9.png">

Where we explicitly declare `foo.opam` as the scope and `bar.ml` as the module.